File: app/src/main/java/com/metahumanz/pacilread/reader/ReaderTitleSpan.java
package com.metahumanz.pacilread.reader;

import android.graphics.Paint;
import android.graphics.Typeface;
import android.text.TextPaint;
import android.text.style.MetricAffectingSpan;

public final class ReaderTitleSpan extends MetricAffectingSpan {
    private final Typeface typeface;
    private final float textSizePx;

    public ReaderTitleSpan(Typeface typeface, float textSizePx) {
        this.typeface = typeface;
        this.textSizePx = textSizePx;
    }

    @Override
    public void updateDrawState(TextPaint textPaint) {
        apply(textPaint);
    }

    @Override
    public void updateMeasureState(TextPaint textPaint) {
        apply(textPaint);
    }

    private void apply(Paint paint) {
        if (paint == null) {
            return;
        }
        if (typeface != null) {
            paint.setTypeface(typeface);
        }
        if (textSizePx > 0f) {
            paint.setTextSize(textSizePx);
        }
    }
}

File: app/src/main/kotlin/com/metahumanz/pacilread/reader/ReaderTitleSpan.kt
package com.metahumanz.pacilread.reader

import android.graphics.Paint
import android.graphics.Typeface
import android.text.TextPaint
import android.text.style.MetricAffectingSpan

class ReaderTitleSpan(private val typeface: Typeface?, private val textSizePx: Float) : MetricAffectingSpan() {
    override fun updateDrawState(textPaint: TextPaint) = apply(textPaint)

    override fun updateMeasureState(textPaint: TextPaint) = apply(textPaint)

    private fun apply(paint: Paint?) {
        if (paint == null) return
        if (typeface != null) paint.typeface = typeface
        if (textSizePx > 0f) paint.textSize = textSizePx
    }
}
